opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -459,6 +471,177 @@ namespace Opm
+                               DeferredLogger& deferred_logger)
+    {
+        const auto& summary_state = ebos_simulator.vanguard().summaryState();
+        bool is_operable = true;
+        bool converged = true;
+        auto& ws = well_state.well(this->index_of_well_);
+        // if well is stopped, check if we can reopen
+        if (this->wellIsStopped()) {
+            this->openWell();
+            auto bhp_target = estimateOperableBhp(ebos_simulator, dt, well_state, summary_state, deferred_logger);
+            if (!bhp_target.has_value()) {
+                // no intersection with ipr
+                const auto msg = fmt::format("estimateOperableBhp: Did not find operable BHP for well {}", this->name());
+                deferred_logger.debug(msg);
+                is_operable = false;
+                // solve with zero rates
+                solveWellWithZeroRate(ebos_simulator, dt, well_state, deferred_logger);
+                this->stopWell();
+            } else {
+                // solve well with the estimated target bhp (or limit)
+                ws.thp = this->getTHPConstraint(summary_state);
+                const double bhp = std::max(bhp_target.value(), prod_controls.bhp_limit);
+                solveWellWithBhp(ebos_simulator, dt, bhp, well_state, deferred_logger);
+            }
+        }
+        // solve well-equation
+        if (is_operable) {
+            converged = this->iterateWellEqWithSwitching(ebos_simulator, dt, inj_controls, prod_controls, well_state, group_state, deferred_logger);
+        }
+
+        const bool isThp = ws.production_cmode == Well::ProducerCMode::THP;
+        // check stability of solution under thp-control
+        if (true) { 
+            if (converged && !this->stopppedOrZeroRateTarget(summary_state, well_state) && isThp) {
+                auto rates = well_state.well(this->index_of_well_).surface_rates;
+                this->adaptRatesForVFP(rates);
+                this->updateIPRImplicit(ebos_simulator, well_state, deferred_logger);
+                bool is_stable = WellBhpThpCalculator(*this).isStableSolution(well_state, this->well_ecl_, rates, summary_state);
+                if (!is_stable) {
+                    // solution converged to an unstable point! 
+                    this->operability_status_.use_vfpexplicit = true;
+                    // msg = ...
+                    auto bhp_stable = WellBhpThpCalculator(*this).estimateStableBhp(well_state, this->well_ecl_, rates, this->getRefDensity(), summary_state);
+                    // if we find an intersection with a sufficiently lower bhp, re-solve equations
+                    const double reltol = 1e-3;
+                    const double cur_bhp = ws.bhp;
+                    if (bhp_stable.has_value() && cur_bhp - bhp_stable.value() > cur_bhp*reltol){
+                        const auto msg = fmt::format("isStableSolution: Well {} converged to an unstable solution, re-solving", this->name());
```

Comment:
Misleading message, it is not written from the isStableSolution() function. Suggest dropping that and keeping the rest of the message.

Reply:
Done